kb/development/python-environment.md

```diff
@@ -0,0 +1,50 @@
+# Setting up a Python environment for Mbed TLS
+
+## Python and package versions
+
+Each branch of Mbed TLS has different minimum requirements on Python. See `README.md` in the branch. Any Python version that's more recent than the minimum should work, however as of November 2022 this is not well tested by the CI so unintended incompatibilities can happen.
+
+Some of the scripts used by Mbed TLS users and developers require third-party packages. Those have version constraints as well. The constraints are documented via `*.requirements.txt` files in the [`scripts` directory](https://github.com/Mbed-TLS/mbedtls/tree/development/scripts).
+
+We run [mypy](http://mypy-lang.org/) and [pylint](https://pylint.pycqa.org/en/latest/) to check Python scripts. It is common that newer versions of Pylint reject our code, because some checks have become stricter or new checks have been added.
+
+## Set up a Python package environment
+
+### Quick setup
+
+Initial setup:
+
+```sh
+python3 -m venv ~/venv-mbedtls-development
+~/venv-mbedtls-development/bin/python scripts/min_requirements.py
```

Comment:
Is this different from
```sh
python3 -m venv ~/venv-mbedtls-development
. ~/venv-mbedtls-development/bin/activate
python scripts/min_requirements.py
```

Reply:
It's the same. Running `activate` in a shell can't really be undone, so I don't want to show instructions to run it without starting a separate shell.

Reply:
Shouldn't `deactivate` undo it?

Reply:
Ah, it should. I'm not in the habit of using it, I don't remember if it's because it used to not exist or it used to not work well or I just don't like the risk when starting a new shell is so easy.